src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -36,32 +36,76 @@ CMachineEvaluation::CMachineEvaluation(CMachine* machine, CFeatures* features,
 	m_features = features;
 	m_labels = labels;
 	m_splitting_strategy = splitting_strategy;
-	m_evaluation_criterion = evaluation_criterion;
+	m_list_evaluation_criterion = list_evaluation_criterion;
+	m_autolock = autolock;
+
+	SG_REF(m_machine);
+	SG_REF(m_features);
+	SG_REF(m_labels);
+	SG_REF(m_splitting_strategy);
+	SG_REF(m_list_evaluation_criterion);
+}
+
+CMachineEvaluation::CMachineEvaluation(CMachine* machine, CFeatures* features,
+		CLabels* labels, CSplittingStrategy* splitting_strategy,
+		CEvaluation* evaluation_criterion, bool autolock)
+{
+    CDynamicObjectArray* container_evaluation_criterion_list = new CDynamicObjectArray();
+    container_evaluation_criterion_list->push_back(evaluation_criterion);
+
+    init();
+
+	m_machine = machine;
+	m_features = features;
+	m_labels = labels;
+	m_splitting_strategy = splitting_strategy;
+	m_list_evaluation_criterion = container_evaluation_criterion_list;
 	m_autolock = autolock;
 
 	SG_REF(m_machine);
 	SG_REF(m_features);
 	SG_REF(m_labels);
 	SG_REF(m_splitting_strategy);
-	SG_REF(m_evaluation_criterion);
+	SG_REF(m_list_evaluation_criterion);
+}
+
+CMachineEvaluation::CMachineEvaluation(CMachine* machine, CLabels* labels,
+		CSplittingStrategy* splitting_strategy,
+		CDynamicObjectArray* list_evaluation_criterion, bool autolock)
+{
+	init();
+
+	m_machine = machine;
+	m_labels = labels;
+	m_splitting_strategy = splitting_strategy;
+	m_list_evaluation_criterion = list_evaluation_criterion;
+	m_autolock = autolock;
+
+	SG_REF(m_machine);
+	SG_REF(m_labels);
+	SG_REF(m_splitting_strategy);
+	SG_REF(m_list_evaluation_criterion);
 }
 
 CMachineEvaluation::CMachineEvaluation(CMachine* machine, CLabels* labels,
 		CSplittingStrategy* splitting_strategy,
 		CEvaluation* evaluation_criterion, bool autolock)
 {
+    CDynamicObjectArray* container_evaluation_criterion_list = new CDynamicObjectArray();
```

Comment:
Since, the user passed a `CEvaluation`, we will have to store it in a list. And the member variable(which is a pointer) that points to that list doesn't point to anything at the construction. I feel we will have to create a new one.